### DISCUSSION TITLE
‼️ Network Version 17 Venus Upgrade Timeline

### DISCUSSION THREAD

**Discussion by Fatman13** [filecoin-project/venus#5386] [Announcements]
**More details on the 🦈 Shark network upgrade can be found [here](https://github.com/filecoin-project/community/discussions/74?sort=top#discussioncomment-3825422)!**

Subscribe to notifications on https://status.filecoin.io/ for any additional network updates!
If you have any concerns or questions, ask in [#fil-venus](https://filecoinproject.slack.com/archives/CEHHJNJS3).

## Venus Releases

**Note: This projected timeline is subject to change based on testing, auditing, new discoveries, and development.**

You will need to be on Venus v1.8.0 or up for the Network v17 Shark upgrade. Version requirement for each Venus component will be listed below as a comment for both calibration and mainnet. Please make sure you use the right version for each Venus services (chain service, storage power service, storage deal service).

## Tentative Releases and Network Upgrade Timelines 

- [x] Calibration compatible RC releases for each Venus service
- [x] 2022-11-07 Calibration-testnet upgrade
- [x] Mainnet compatible releases for each Venus service
- [x] 2022-11-30, mainnet upgrade

## Resources

- [ ] https://github.com/filecoin-project/venus/discussions/5368
- [ ] https://github.com/filecoin-project/venus/issues/5287
- [ ] https://github.com/filecoin-project/venus/issues/5311

**Comment by Fatman13** [comment]
## Calibration compatible RC releases

Please use the following Veuns services version for pre-NV17 upgrade testing on Calibration network.

### Chain Service

- venus ~~[v1.8.0-rc3](https://github.com/filecoin-project/venus/releases/tag/v1.8.0-rc3)~~ [v1.8.0-rc4](https://github.com/filecoin-project/venus/releases/tag/v1.8.0-rc4)
- venus-auth ~~[v1.8.0-rc1](https://github.com/filecoin-project/venus-auth/releases/tag/v1.8.0-rc1)~~ [v1.8.0-rc4](https://github.com/filecoin-project/venus-auth/releases/tag/v1.8.0-rc4)
- venus-messager  ~~[v1.8.0-rc2](https://github.com/filecoin-project/venus-messager/releases/tag/v1.8.0-rc2)~~ [v1.8.0-rc3](https://github.com/filecoin-project/venus-messager/releases/tag/v1.8.0-rc3)
- venus-gateway ~~[v1.8.0-rc1](https://github.com/ipfs-force-community/venus-gateway/releases/tag/v1.8.0-rc1)~~ [v1.8.0-rc2](https://github.com/ipfs-force-community/venus-gateway/releases/tag/v1.8.0-rc2)
- venus-miner ~~[v1.8.0-rc3](https://github.com/filecoin-project/venus-miner/releases/tag/v1.8.0-rc3)~~ [v1.8.0-rc5](https://github.com/filecoin-project/venus-miner/releases/tag/v1.8.0-rc5)
- venus-wallet ~~[v1.8.0-rc2](https://github.com/filecoin-project/venus-wallet/releases/tag/v1.8.0-rc2)~~ [v1.8.0-rc3](https://github.com/filecoin-project/venus-wallet/releases/tag/v1.8.0-rc3)

### Storage Power Service
- venus-cluster ~~[v0.4.0-rc3-nv17](https://github.com/ipfs-force-community/venus-cluster/tree/v0.4.0-rc3-nv17)~~ [v0.4.0-rc4](https://github.com/ipfs-force-community/venus-cluster/tree/v0.4.0-rc4)
- venus-cluster  ~~[v0.5.0-beta1-nv17](https://github.com/ipfs-force-community/venus-cluster/tree/v0.5.0-beta1-nv17)~~ [v0.5.0-beta2](https://github.com/ipfs-force-community/venus-cluster/tree/v0.5.0-beta2)

### Storage Deal Service

- venus-market ~~[v2.4.0-rc1](https://github.com/filecoin-project/venus-market/releases/tag/v2.4.0-rc1)~~    ~~[v2.4.0-rc2](https://github.com/filecoin-project/venus-market/releases/tag/v2.4.0-rc2)~~   ~~[v2.4.0-rc3](https://github.com/filecoin-project/venus-market/releases/tag/v2.4.0-rc3)~~ [v2.4.0-rc4](https://github.com/filecoin-project/venus-market/releases/tag/v2.4.0-rc4)

**Comment by Fatman13** [comment]
## Mainnet compatible releases 

Please use the following Venus services version for NV17 mainnet.

### Chain Service

- venus [v1.8.0](https://github.com/filecoin-project/venus/releases/tag/v1.8.0)
- venus-auth [v1.8.0](https://github.com/filecoin-project/venus-auth/releases/tag/v1.8.0)
- venus-messager [v1.8.0](https://github.com/filecoin-project/venus-messager/releases/tag/v1.8.0)
- venus-gateway [v1.8.0](https://github.com/ipfs-force-community/venus-gateway/releases/tag/v1.8.0)
- venus-miner [v1.8.0](https://github.com/filecoin-project/venus-miner/releases/tag/v1.8.0)
- venus-wallet [v1.8.0](https://github.com/filecoin-project/venus-wallet/releases/tag/v1.8.0)

### Storage Power Service

- venus-cluster [v0.4.0-rc5](https://github.com/ipfs-force-community/venus-cluster/tree/v0.4.0-rc5)
- venus-cluster [v0.5.0-beta3](https://github.com/ipfs-force-community/venus-cluster/tree/v0.5.0-beta3)

### Storage Deal Service

- venus-market [v2.4.0](https://github.com/filecoin-project/venus-market/releases/tag/v2.4.0)

## NV17 upgrade guide

- nv17 upgrade EN guide: https://venus.filecoin.io/operation/nv17-upgrade.html
- nv17 upgrade [CN guide](https://github.com/filecoin-project/venus/discussions/5386#discussioncomment-4163911): https://venus.filecoin.io/zh/operation/nv17-upgrade.html

**Comment by simlecode** [comment]
## nv17 各组件升级文档

如果 `venus-wallet` 或者 `venus-gateway` 的版本从 `v1.6.x` 升级到 `v1.8.0`，请先阅读 [Venus组件v1.7.0使用注意事项 ](https://github.com/filecoin-project/venus/discussions/5265)，注意 `venus-gateway` 和 `venus-wallet`的升级顺序。

### 时间线

升级时间：2022-11-30T14:00:00Z，北京时间是 2022-11-30 22:00:00

### 各组件版本

* [calibrationnet](https://github.com/filecoin-project/venus/discussions/5386#discussioncomment-3893446)
* [mainnet](https://github.com/filecoin-project/venus/discussions/5386#discussioncomment-3893449)

### venus-auth

影响功能：

依赖升级组件：无

注意事项：

1. `miners` 表增加字段 `open_mining`，0 表示不出块，1 表示出块，默认是 1


### venus

影响功能：支持nv17网络

依赖升级组件:无

注意事项：

编译：先 `make dist-clean` 再 `make`，这样可以防止未能正常升级 `filecoin-ffi` 而带来的问题

1. 升级后检查vk文件是否完整

2. 升级后检查配置文件，升级高度是否正常设置，配置文件默认在 `~/.venus/config.json` :
    ```
    # cali
    "upgradeSkyrHeight": 16800
    # mainnet
    "upgradeSkyrHeight": 2383680
    ```

3. 升级后需检查是否正常同步区块

4. 检查主网 v9 actors code，确保与下面输出一样

    ```
    ./venus state actor-cids --network-version 17

    # 顺序可能会不一样
    Network Version: 17
    Actor Version: 9
    Actor             CID
    reward            bafk2bzacebezgbbmcm2gbcqwisus5fjvpj7hhmu5ubd37phuku3hmkfulxm2o
    system            bafk2bzaceagvlo2jtahj7dloshrmwfulrd6e2izqev32qm46eumf754weec6c
    _manifest         bafy2bzaceb6j6666h36xnhksu3ww4kxb6e25niayfgkdnifaqi6m6ooc66i6i
    storagemarket     bafk2bzacec3j7p6gklk64stax5px3xxd7hdtejaepnd4nw7s2adihde6emkcu
    multisig          bafk2bzacec4va3nmugyqjqrs3lqyr2ij67jhjia5frvx7omnh2isha6abxzya
    paymentchannel    bafk2bzacebhdvjbjcgupklddfavzef4e4gnkt3xk3rbmgfmk7xhecszhfxeds
    storagepower      bafk2bzacedsetphfajgne4qy3vdrpyd6ekcmtfs2zkjut4r34cvnuoqemdrtw
    verifiedregistry  bafk2bzacecf3yodlyudzukumehbuabgqljyhjt5ifiv4vetcfohnvsxzynwga
    datacap           bafk2bzacebb6uy2ys7tapekmtj7apnjg7oyj4ia5t7tlkvbmwtxwv74lb2pug
    account           bafk2bzacect2p7urje3pylrrrjy3tngn6yaih4gtzauuatf2jllk3ksgfiw2y
    cron              bafk2bzacebcec3lffmos3nawm5cvwehssxeqwxixoyyfvejy7viszzsxzyu26
    init              bafk2bzacebtdq4zyuxk2fzbdkva6kc4mx75mkbfmldplfntayhbl5wkqou33i
    storageminer      bafk2bzacedyux5hlrildwutvvjdcsvjtwsoc5xnqdjl73ouiukgklekeuyfl4
    ```

5. 若不排查问题，rust日志级别**不建议**设置为 `trace`，因为会打印较多日志

6. 升级后可以通过命令来 `./venus state get-actor t01000` 来确认是否升级成功


### venus-messager

功能：

1. 增加对basefee的控制，可以对某个地址单独设置 basefee，当链上basefee高于限制的basefee，消息将不会被推送出去

依赖升级组件: 

注意事项：

1. 升级后观察消息能否正常上链


### venus-miner

影响功能:

依赖升级组件:

注意事项: 

1. 从 `1.7.*`升级到`v1.8.0`会自动迁移配置文件，从 `1.6.*` 升级到`v1.8.0`需重新初始化`repo`(init)，[参考](https://github.com/filecoin-project/venus-miner/blob/master/docs/zh/CHANGELOG.md#%E6%B3%A8%E6%84%8F%E4%BA%8B%E9%A1%B9-2)
2. 升级后能否正常出块


### venus-gateway

影响功能：

注意事项：

编译：先 `make dist-clean` 再 `make`

1. 升级后检查钱包：`./venus-gateway wallet list`
2. 升级后检查矿工：`./venus-gateway miner list`


### venus-market

影响功能：

注意事项：

编译：先 `make dist-clean` 再 `make`

### venus-wallet

影响功能：

注意事项：

1. 启动后需要解锁钱包
2. 是否注册到gateway

**Reply by Fatman13** [reply]
66666，我待会给他传到venus-docs上面。

**Reply by Fatman13** [reply]
https://venus.filecoin.io/zh/operation/nv17-upgrade.html
